hooks/src/lib/hooks/useCardBrand.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+
+import { AMEX, DINERS, MASTER, UNION, VISA } from '../constants';
+import { Brand, BrandInfo } from '../types/card';
+
+export default function useCardBrand({ cardNumbers }: { cardNumbers: string }) {
+  const [brand, setBrand] = useState<Brand>(null);
+  /**
+   * 카드 번호에 카드 브랜드에 적합한 번호가 존재하는 지 여부를 판단하는 함수
+   * @param brandNumbers : 카드 브랜드에 속하는 번호들
+   */
+  const isBrandNumbers = (brandNumbers: number[]) => {
+    const prefixNumberCount = brandNumbers[0].toString().length;
+    const prefix = Number(cardNumbers.slice(0, prefixNumberCount));
+
+    return brandNumbers.includes(prefix);
+  };
+
+  /**
+   * 카드 번호에 맞는 카드 브랜드를 반환, 만약 해당하는 카드 브랜드가 없으면 undefined를 반환
+   * @param brandInfo : 카드 브랜드 정보
+   * @param isValidatedNumbers  : 카드 번호가 카드 브랜드 번호에 속하는 지 여부
+   */
+  const getBrand = <T>(brandInfo: BrandInfo<T>, isValidatedNumbers: boolean): Brand | undefined => {
+    const { length, name } = brandInfo;
+
+    if (cardNumbers.length !== length) return;
+    if (isValidatedNumbers) {
+      return name;
+    }
+  };
```

Comment:
brandInfro가 numbers를 가지고 있기 때문에 굳이 밖에서 validation을 할 필요는 없어보입니다.

```suggestion
  const getBrand = <T>(brandInfo: BrandInfo<T>): Brand | undefined => {
    const { length, name, numbers } = brandInfo;

    if (cardNumbers.length !== length) return;
    if (isBrandNumbers(numbers)) {
      return name;
    }
  };
```